low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -269,9 +270,53 @@ def prepare_mem_json(self, status, description):
         }
         return memory_dict
 
+    def get_fanmodules_info(self):
+        """Get the Fan sensor information using ipmitool"""
+        data = []
+        fan_count = 1
+        sensor_reading = self._ipmi.get_sensor_list_by_type('Fan')
+        for reading in sensor_reading:
+            uid = 'fan_' + str(fan_count)
+            fan_resp = self.format_ipmi_fan_reading(reading, uid)
+            data.append(fan_resp)
+            fan_count += 1
+        return data
+
+    def format_ipmi_fan_reading(self, fan_reading, uid):
+        """builds json resposne from ipmi tool response."""
+        sensor_id = fan_reading[0]
+        health_desc = "good"
+        recommendation = "NA"
+        sensor_props = self._ipmi.get_sensor_props(sensor_id)
+        status = 'OK' if fan_reading[2] == 'ok' else 'NA'
+        if status != 'OK':
+            health_desc = "bad"
+            recommendation = sspl_constants.DEFAULT_ALERT_RECOMMENDATION
+        lower_critical = sensor_props[1].get('Lower Critical', 'NA')
+        upper_critical = sensor_props[1].get('Upper Critical', 'NA')
+
+        resp = {
+            "uid": uid,
```

Comment:
As per the [comment](https://github.com/Seagate/cortx-monitor/pull/426/files#r653582700,) on #426 PR, Please use 'get_health_template' and 'set_health_data' function.

Reply:
Addressed @shriya-deshmukh.
Please review now